third-culture-software/bhima
# Minor improvements to the Annual Clients Report.

This PR implements the following changes

- Minor improvement to the Debtor Group Management page
- Minor improvements to the Annual Clients Report

*TESTING* 

- Use the bhima_test or a production database for testing this part
- Debtor Group Management page
   - Go to Administration > Debtor Group Management
      - Edit one debtor group and add a note
      - The note will be shown for for each debtor group that has one

- Annual Clients Report
   - Use a production database to test this part
   - Go to the Finances > Report > Annual Clients Report page
   - Select various options and click on the [Preview] button
   - Try the various combinations of the check boxes
     table
	 
        | Hide Locked Clients | Include Cash Clients | Cash Col? | Locked Col? |
        | --------------------|----------------------|-----------|-------------|
        |  NO                 | No                   | No        | No*         |
        |  NO                 | Yes                  | Shown     | No*         |
        |  Yes                | No                   | No        | No          |
        |  Yes                | Yes                  | Show      | No          |
	 
	 * If you chose "NO" for the "Hide locked clients" option,
       note that if no locked accounts are shown, it will print
	   a message about that at the bottom of the page.

   - To see the normal "Locked" go the the Administrion > Debtor Groups
     Management page and lock the "NGO IMA World Health" group
   - Repeat the verification process and note that a "Locked" 
     column will appear if the locked debtor groups are shown and
	 no message will appear at the end since there is a locked 
	 debtor group.

        | Hide Locked Clients | Include Cash Clients | Cash Col? | Locked Col? |
        | --------------------|----------------------|-----------|-------------|
        |  NO                 | No                   | No        | Shown       |
        |  NO                 | Yes                  | Shown     | Shown       |
        |  Yes                | No                   | No        | No          |
        |  Yes                | Yes                  | Shown     | No          |

Closes https://github.com/IMA-WorldHealth/bhima/issues/6170
